src/Compiler/FSComp.txt

```diff
@@ -1646,3 +1650,11 @@ reprStateMachineInvalidForm,"The state machine has an unexpected form"
 3522,tcAnonRecdDuplicateFieldId,"The field '%s' appears multiple times in this record expression."
 3523,tcAnonRecdTypeDuplicateFieldId,"The field '%s' appears multiple times in this anonymous record type."
 3524,parsExpectingExpressionInTuple,"Expecting expression"
+3530,tcTraitIsStatic,"Trait '%s' is static"
+3531,tcTraitIsNotStatic,"Trait '%s' is not static"
+3532,tcTraitMayNotUseComplexThings,"A trait may not specify optional, in, out, ParamArray, CallerInfo or Quote arguments"
+3533,tcInvalidSelfConstraint,"Invalid constraint. Valid constraint forms include \"'T :> ISomeInterface\" for interface constraints and \"SomeConstrainingType<'T>\" for self-constraints. See https://aka.ms/fsharp-type-constraints."
+3534,tcTraitInvocationShouldUseTick,"Invocation of a static constraint should use \"'T.Ident\" and not \"^T.Ident\", even for statically resolved type parameters."
+3535,tcUsingInterfacesWithStaticAbstractMethods,"Declaring \"interfaces with static abstract methods\" is an advanced feature. See https://aka.ms/fsharp-iwsams for guidance. You can disable this warning by using '#nowarn \"3535\"' or '--nowarn:3535'."
+3536,tcUsingInterfaceWithStaticAbstractMethodAsType,"'%s' is normally used as a type constraint in generic code, e.g. \"'T when ISomeInterface<'T>\" or \"let f (x: #ISomeInterface<_>)\". See https://aka.ms/fsharp-iwsams for guidance. You can disable this warning by using '#nowarn \"3536\"' or '--nowarn:3536'."
```

Comment:
The aka.ms link isn't working yet. Where should it point to?

Reply:
We will add it before release, pointing either to the RFC or a new doc page